addons/base_automation/models/base_automation.py

```diff
@@ -255,14 +256,15 @@ def _process(self, records, domain_post=None):
                         'domain_post': domain_post,
                     }
                     try:
-                        self.action_server_id.sudo().with_context(**ctx).run()
+                        self.action_server_id.with_context(**ctx).run()
                     except Exception as e:
                         self._add_postmortem_action(e)
                         raise e
 
     def _check_trigger_fields(self, record):
         """ Return whether any of the trigger fields has been modified on ``record``. """
-        if not self.trigger_field_ids:
+        self_sudo = self.sudo()
```

Comment:
Seems unnecessary, `_process` is the only caller so this would only be required if `_process` is not sudoed, in which case wouldn't its field access (to `action_server_id`) fail? Or does that magically work because m2o?

Reply:
Same, without this sudo the below test fails when trying to read the content of `trigger_field_ids`